homeassistant/components/uvc/camera.py

```diff
@@ -217,6 +217,7 @@ def disable_motion_detection(self):
         channels = caminfo["channels"]
         for channel in channels:
             if channel["isRtspEnabled"]:
-                return channel["rtspUris"][0]
+                hostRtspIndexes = [i for i, item in enumerate(channel["rtspUris"]) if re.search(self._nvr._host, item)]
```

Comment:
We can use `next` with a generator expression since we only want the first item.

```py
uri = next((
    uri for i, uri in enumerate(channel["rtspUris"])
    if re.search(self._nvr._host, item)
))
return uri
```

If we think it might not find any item at all, we can also set a default.

Reply:
Thanks for this @MartinHjelmare ! I'm not near my home assistant install for the next week so I'll make this change and test it in a weeks time or so.